packages/headless/src/features/facets/category-facet-set/category-facet-utils.ts

```diff
@@ -33,16 +33,46 @@ export function partitionIntoParentsAndValues<
   return {parents, values};
 }
 
-export function getActiveValueFromValueTree(
-  valuesAsTrees: CategoryFacetValue[]
-): CategoryFacetValue | undefined {
+export function findActiveValueAncestry<T extends GenericCategoryFacetValue>(
+  valuesAsTree: T[]
+): T[] {
+  const ancestryMap = new Map<T, T>();
+  const activeValue = getActiveValueFromValueTree(valuesAsTree, ancestryMap);
+  return activeValue ? getActiveValueAncestry(activeValue, ancestryMap) : [];
+}
+
+function getActiveValueFromValueTree<TValue extends GenericCategoryFacetValue>(
+  valuesAsTrees: TValue[],
+  ancestryMap?: Map<TValue, TValue>
+): TValue | undefined {
   const valueToVisit = [...valuesAsTrees];
   while (valueToVisit.length > 0) {
     const currentValue = valueToVisit.shift()!;
     if (currentValue.state === 'selected') {
       return currentValue;
     }
-    valueToVisit.unshift(...currentValue.children);
+    if (ancestryMap) {
```

Comment:
Yes, but good catch 🥇 
Both your comments are because I initially left the `getActiveValueFroMValueTree` exposed, and didn't want to break its signature.
This is not the case any more so:
 - I can be sure that the ancestryMap exist
 - I can return it instead